crates/client/mapping-sync/src/sync_blocks.rs

```diff
@@ -88,14 +101,19 @@ where
         block_hash: substrate_block_hash,
         starknet_block_hash: block_hash.into(),
         starknet_transaction_hashes: Vec::new(),
+        cache,
     };
 
     backend.mapping().write_hashes(mapping_commitment)?;
 
     Ok(())
 }
 
+/// # Arguments
+///
+/// - `cache`: whether more information should be cached when storing the block in the database.
 fn sync_one_block<B: BlockT, C, BE, H>(
+    cache: bool,
```

Comment:
Rather than passing this arg in every intermediate func.

pass it to `MadaraBlackend::open` in `services.rs`
```rust
    let madara_backend = Arc::new(MadaraBackend::open(&config.database, &db_config_dir(config))?);
```

add it to this struct:
```rust
pub struct MappingDb<B: BlockT> {
    pub(crate) db: Arc<dyn Database<DbHash>>,
    pub(crate) write_lock: Arc<Mutex<()>>,
    pub(crate) _marker: PhantomData<B>,
}
```

So now you can access it any time you want

Reply:
That's what I had asked you; you told me to pass it through the `MappingCommitment`. I think we misunderstood ourselves. I'll change this.